hangman.py

```diff
@@ -0,0 +1,173 @@
+            print('  |')
+
+        if wrongGuesses == 6:  # printing third line of gallows for all states
+            print('  |      /|\\')
+        elif wrongGuesses == 5:
+            print('  |      /|')
+        elif wrongGuesses >= 2:
+            print('  |       |')
+        else:
+            print('  |')
+
+        if wrongGuesses >= 4:  # printing fourth line of gallows for all states
+            print('  |      / \\')
+        elif wrongGuesses == 3:
+            print('  |      / ')
+        else:
+            print('  |')
+
+        print('__|____\n')  # printing fifth line of gallows for all states
+
+    def newGame(self):
+
+        self.openFile()
+        self.wordPick()
+
+        letters = list(string.ascii_lowercase)
+        guessed = copy.deepcopy(letters)
+        guesses = ''
+        target = ''
+        output = ''
+        stateHM = 0
+
+        for char in self.word:
+            target += ' ' + char + ' '
+
+        for char in self.word:
+            output += ' _ '
+        print(output)
+
+        while stateHM < 6:
+            output = ''
+
+            print(*guessed, sep=' ')
+            guess = input("Choose a letter: ")
+            guess = guess.lower()
+
+            if guess in letters:
+                if guess in guessed:
+                    guessed.remove(guess)
+
+                if guess not in guesses:
+                    guesses += guess
+
+                    if guess not in self.word:
+                        print(f'{guess} is not in the word!')
+                        stateHM += 1
+                    else:
+                        print(f'{guess} is in the word!')
+
+                else:
+                    print(f'You already guessed "{guess}"!')
+
+                time.sleep(1)
+                self.printhangman(stateHM)
+
+                for char in self.word:
+                    if char in guesses:
+                        output += ' ' + char + ' '
+                    else:
+                        output += ' _ '
+
+                print(output)
+
+                if output == target:
+                    print(f'You won in {len(guesses)} guesses!')
+                    break
+                elif stateHM == 6:
+                    print(f'You lose!\nThe word was {self.word}')
+                    break
+            elif len(guess) != 1:
+                print("Please pick one letter from the list...")
+            else:
+                print(f'"{guess}" is not a letter! Try again...')
+
+            time.sleep(.75)
+
+
+        again = input('Play again? (y/n)')
+        if again == 'y':
+            self.newGame()
+        else:
+            print("Thanks for playing!")
+
+def main():
+    # driver function if hangman.py is run
+
+    # the next 3 lines of code setup the CLI input using the argparse library
+    parser = argparse.ArgumentParser(description='Play a game of Hangman')
+    parser.add_argument('-l', '--minlength', type=int, metavar='', required=True,
```

Comment:
we may not want to use a lowercase "L" since it looks like a 1 in the terminal, this could confuse the user. Maybe "-m"?